lib/datadog/statsd/serialization/stat_serializer.rb

```diff
@@ -38,7 +38,11 @@ def global_tags
         attr_reader :tag_serializer
 
         def formatted_metric_name(metric_name)
-          formatted = Symbol === metric_name ? metric_name.name : metric_name.to_s
+          if Symbol === metric_name && metric_name.respond_to?(:name)
```

Comment:
Minor: Since this if will always be true or false for a given Ruby version, it may be worth introducing a new method that's conditionally defined for that Ruby; e.g. something like

```ruby
if RUBY_VERSION < '3'
  def metric_name_to_string(metric_name)
    Symbol === metric_name ? metric_name.name : metric_name.to_s
  end
else
  def metric_name_to_string(metric_name)
    metric_name.to_s
  end
end
```

...this way each Ruby gets its own correct version of the code and no need to keep checking.